icu4c/source/i18n/number_output.cpp

```diff
@@ -16,6 +16,7 @@ namespace number {
 
 UPRV_FORMATTED_VALUE_SUBCLASS_AUTO_IMPL(FormattedNumber)
 
+#define UPRV_NOARG
```

Comment:
I used UPRV_NOARG instead of NOARG in the off-change that the macro name might collide with somebody else.

Reply:
If you wanted to be pedantic, you could also `#undef UPRV_NOARG` when finished using it, but that doesn't really matter since this is a CPP file.  (I am not advocating that you change anything.)